performance/api_load_test.go

```diff
@@ -0,0 +1,212 @@
+package performance
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/internal/cltest"
+	"github.com/stretchr/testify/assert"
+	"github.com/tsenart/vegeta/lib"
+)
+
+type Requirement struct {
+	requestsPerSecond int
+	averageLatencyMax time.Duration
+	percentileMax     time.Duration
+	runDurationMax    time.Duration
+}
+
+var (
+	r            Requirement
+	requirements = []Requirement{
+		{20, time.Millisecond * 100, time.Millisecond * 100, time.Millisecond * 100},
+		{40, time.Millisecond * 200, time.Millisecond * 200, time.Millisecond * 200},
+		{60, time.Millisecond * 300, time.Millisecond * 300, time.Millisecond * 300},
+	}
+
+	app     *cltest.TestApplication
+	tests   func(t *testing.T)
+	cleanup func()
+)
+
+func Setup() {
+	tests = func(t *testing.T) {
+		for _, requirement := range requirements {
+			app, cleanup = cltest.NewApplicationWithKeyStore()
+			r = requirement
+			t.Log(fmt.Printf("Requests per second: %d\n", r.requestsPerSecond))
+			t.Run("CreateJobs", CreateJobApiUnderLoad)
+			t.Run("ViewJobs", ViewJobApiUnderLoad)
+			t.Run("CreateJobRuns", CreateJobRunsApiUnderLoad)
+			t.Run("ViewJobRuns", ViewJobRunsApiUnderLoad)
+			cleanup()
+		}
+	}
+	return
+}
+
+func TestNodeUnderLoad(t *testing.T) {
+	Setup()
+	tests(t)
+}
+
+func CreateJobApiUnderLoad(
+	t *testing.T,
+) {
+	Setup()
```

Comment:
It seems that `Setup` calls `CreateJobApiUnderLoad` and here we are with `CreateJobApiUnderLoad` calling `Setup`. You've run this to completion so there isn't a recursive loop but it does make us question the layout.